### PR TITLE
txn: add description about deprecating the tidb_disable_txn_auto_retry system variable 

### PR DESCRIPTION
**Pull request by cfzjywxk**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/docs-cn/pull/16592

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: optimistic-transaction.md [hunk 1]
```diff
@@ -65,6 +65,10 @@ aliases: ['/docs-cn/dev/optimistic-transaction/','/docs-cn/dev/reference/transac
 
 ## 事务的重试
 
+> **注意：**
+>
+> 从 v8.0.0 开始，`tidb_disable_txn_auto_retry` 被废弃，推荐使用 [悲观事务模式](/pessimistic-transaction.md)。如使用乐观事务模式发生冲突时，请在应用里捕获错误并重试。
```
Comment (lilin90) [inline optimistic-transaction.md:70]:
```suggestion
> 从 v8.0.0 开始，`tidb_disable_txn_auto_retry` 被废弃，不再支持乐观事务的自动重试。推荐使用 [悲观事务模式](/pessimistic-transaction.md)。如果使用乐观事务模式发生冲突，请在应用里捕获错误并重试。
```

**Review by qiancai — APPROVED** [review]

File: optimistic-transaction.md [hunk 2]
```diff
@@ -75,7 +79,7 @@ aliases: ['/docs-cn/dev/optimistic-transaction/','/docs-cn/dev/reference/transac
 # 设置是否禁用自动重试，默认为 “on”，即不重试。
 tidb_disable_txn_auto_retry = OFF
 # 控制重试次数，默认为 “10”。只有自动重试启用时该参数才会生效。
-# 当 “tidb_retry_limit= 0” 时，也会禁用自动重试。
+# 当 “tidb_retry_limit= 0” 时，也会禁用自动重试。
```
Comment (MyonKeminta) [inline optimistic-transaction.md:82]:
```suggestion
# 当 “tidb_retry_limit = 0” 时，也会禁用自动重试。
```

**Review by MyonKeminta — APPROVED** [review]
Rest LGTM

**Review by lilin90 — APPROVED** [review]
/lgtm

**Pull request merged**